File: AoC2015/JavaScript/day2.js
//puzzle setup
const fs = require('fs');
var input = fs.readFileSync('./day2in.txt', 'utf8');
input = input.split(/\r?\n/);

//do stuff
var sum = 0;
for (i in input) input[i] = input[i].split('x').map(Number);

for (let i = 0; i < input.length; i++)
{
    let temp = [];
    temp.push(input[i][0] * input[i][1]); // l*w
    temp.push(input[i][1] * input[i][2]); // w*h
    temp.push(input[i][2] * input[i][0]); // h*l
    let small = Math.min(...temp);
    sum += small;
    sum += temp.reduce((a,b)=>a+b,0) * 2;
}
console.log(sum);

File: AoC2015/Python3/day2.py
import numpy

#puzzle setup
with open('./day2in.txt','r') as pInput:
    pInput = pInput.read().splitlines(0)

#do stuff
mysum = 0
pInput = [i.split('x') for i in pInput]

for x in pInput:
    x = [int(i) for i in x]
    temp = []
    temp.append(x[0] * x[1])
    temp.append(x[0] * x[2])
    temp.append(x[1] * x[2])
    mysum += min(temp)
    mysum += numpy.sum(temp) * 2

print(mysum)
